builder/comp-builder.nix

```diff
@@ -356,6 +356,7 @@ let
       description = package.synopsis or "";
       license = haskellLib.cabalToNixpkgsLicense package.license;
       platforms = if platforms == null then lib.platforms.all else platforms;
+      mainProgram = if haskellLib.isExecutableType componentId then exeName else "";
```

Comment:
Can we just not include this attribute if it's not an executable? What does nix run do if it's the empty string as opposed to missing?

Worth a comment explaining what you said in the PR about nix run so we know what this is for in future.

Reply:
Is there something we can put there that would make it actually fail? What happens if we put `error "Component <component> is not an executable, you can't run it"`?

Reply:
I'm not sure what the idiomatic thing to do here is!

Reply:
Setting it as
```
mainProgram = if haskellLib.isExecutableType componentId then exeName
              else throw "Component ${name} is not an executable, you can't run it.";
```
allows `nix run` to work on executables while showing that error message for non-executable components like `lib`s, but I'm not sure if this is a good idea since `[package].meta ? mainProgram` now returns `true` and the `[package].meta.mainProgram or [something else]` pattern no longer works. This may or may not be desirable, and I have no clue what the idiomatic thing to do here is either.

Reply:
Fair enough. It just seems a bit sad that the fallback case is "try and run an obviously non-existent executable" rather than "tell the user they can't do that".

Reply:
Yeah. But I suppose that's true of any nix derivations independently from haskell.nix, and the iffiness comes from nixpkgs not having `meta.mainProgram` before but having it now, and wanting to support derivations that depend on old behaviour.